.devcontainer/cuda12.4-llvm16/devcontainer.json

```diff
@@ -31,7 +31,9 @@
     "vscode": {
       "extensions": [
         "llvm-vs-code-extensions.vscode-clangd",
-        "xaver.clang-format"
+        "xaver.clang-format",
+        "nvidia.nsight-vscode-edition",
+        "ms-vscode.cmake-tools"
```

Comment:
question: What does `ms-vscode.cmake-tools` do for us?

Reply:
As we are trying to make the process as user-friendly and seamless as possible the extension allows us to configure with the desired preset (all-dev-debug in our case), build and run from the UI so the user doesn't have to use the terminal.

Reply:
Cool! I look forward to seeing a demo!